### PR TITLE
making the main blog page editable with Gutenberg (above article list)

### PR DESCRIPTION
**Pull request by eric-michel**
@dswebsme, here's my first shot at this. It's fairly straightforward, although I had to do some slightly hacky bits to make it work:
- The block editor does not appear for the blog page unless content already exists for that page. The only way to get it to appear was to essentially simulate a comment that would make the editor think content always exists for that particular page. Fortunately it won't actually save, since it's an HTML comment.
- For the page title, `is_singular` wouldn't give me the appropriate title format for the main blog page since the query returned multiple results, even after using `setup_postdata` to pull only the content from the main blog page record. So I added another conditional in the `if` statement to check if it was the main blog page.

 Overall I think it's worth adding into core, as it would be super nice to be able to put whatever content you wanted above your list of articles without template modifications.

One additional feature I'd love to add is a simple notice at the top of the page that states that the blog listing will appear below the content in the editor, so that the user will know this is the blog page they're editing. Other than that, I think this is pretty much ready for prime time, pending your review.

Thanks for taking a look!

### CODE REVIEW

File: home.php [hunk 1]
```diff
@@ -16,31 +16,13 @@
...
-        if( has_post_thumbnail( get_option( 'page_for_posts', true ) ) ) :
-          ?>
-
-          <div class="page-thumbnail">
-            <?php echo get_the_post_thumbnail( get_option( 'page_for_posts', true ) ); ?>
-          </div><!-- .post-thumbnail .page-thumbnail -->
-
-          <?php
-        endif;
-        ?>
```
Comment (dswebsme) [inline home.php:40]: @eric-michel We need to test this. In scenarios where the `posts` page is not the front page it looks like we run the risk of not having a `<header class="page-header">` element on those pages which I believe is a persistent element across all other templates (again, need to test to verify).

I worry that by removing it we'll introduce a template with a unique structure that could affect layout / styling.

File: functions.php [hunk 2]
```diff
@@ -569,3 +569,15 @@ function emma_cart_anchor_fragment( $fragments ) {
...
+ * Simulate non-empty content to enable Gutenberg editor on the main blog page (kind of hacky, but the only way I could find to do it)
+ */
+function emma_enable_gutenberg_editor_for_blog_page( $replace, $post ) {
+  if ( ! $replace && absint( get_option( 'page_for_posts' ) ) === $post->ID && empty( $post->post_content ) ) {
+    // This comment will be removed by Gutenberg since it won't parse into block.
+    $post->post_content = '<!--non-empty-content-->';
+  }
+  return $replace;
+}
+add_filter( 'replace_editor', 'emma_enable_gutenberg_editor_for_blog_page', 10, 2 );
```
Comment (dswebsme) [inline functions.php:583]: I'm ok with this if there aren't any well supported alternatives. I'll take a look locally and see how this behaves in a real world scenario and provide additional feedback soon.

**Review by dswebsme — COMMENTED** [review]
@eric-michel See in-line comments.

File: home.php (hunk 1 above)
Comment (eric-michel) [reply home.php:40]: @dswebsme in looking at our normal page template, it doesn't look like a `<header class="page-header">` is ever generated. It's always `entry-header`, and only if the title is not hidden. This brings the main blog page in line with those templates, if I'm looking at things right.

Comment (dswebsme) [reply home.php:40]: The traditional separation between `page` and `post` in templates presents itself in cases like this where `.page-header` and `.entry-header` are used respectively. Archive pages can get particularly interesting such that the overall template utilizes `.page-header` while individual posts rendered by the loop might have their own `.entry-header` associated with each post.

When I search the code for `page-header` I find about what I'd expect to see in `emma`:
page-header

Comment (dswebsme) [reply home.php:40]: ^ Notice that `single.php` is not on this list as it utilizes `entry-header` in its' related template parts, namely `content.php`.

```<div class="entry-content">```

Comment (dswebsme) [reply home.php:40]: Upon further investigation is looks like there is some inconsistency in how / where we use `page-header` vs. `entry-header` so I'm resolving this conversation so we can address that in a separate issue.

**Pull request merged**